canvas/services/event.py

```diff
@@ -52,6 +44,29 @@ def get_question_stats(question):
                 answers[answer] = 0
             answers[answer] += 1
 
+    users = []
+    progress_submission_statuses = [0, 0, 0]
+    student_submissions = submissions.filter(uqj__user__role="Student")
+    correct_submissions = student_submissions.filter(is_correct=True).values("uqj__user_id")
+    for correct in correct_submissions:
+        if not users.__contains__(correct["uqj__user_id"]):
+            progress_submission_statuses[0] += 1
+            users.append(correct["uqj__user_id"])
+
+    partially_correct_submissions = student_submissions.filter(is_partially_correct=True).values("uqj__user_id")
```

Comment:
nit;

For speed, we could adjust the filter so it's already ignoring the submissions from students with ids already in the `users` list.

Something like 
```py
student_submissions.filter(is_partially_correct=True).exclude(uqj__user_id__in=users).values("uqj__user_id")
```
_Note the `.exclude` and `__in`